lib/spack/spack/solver/concretize.lp

```diff
@@ -1187,11 +1327,14 @@ attr("node_compiler_version_satisfies", node(ID, Package), Compiler, Version) :-
 % Check whether the DAG has any built package
 has_built_packages() :- build(X), not external(X).
 
-% If we build packages, the runtime nodes must use an available compiler
-1 { node_compiler(PackageNode, CompilerID) : build(PackageNode), not external(PackageNode) } :-
-  has_built_packages(),
-  runtime(RuntimePackage),
-  node_compiler(node(_, RuntimePackage), CompilerID).
+% "gcc-runtime" is always built
+:- concrete(node(X, "gcc-runtime")), has_built_packages().
+
+% The "gcc" linked to "gcc-runtime" must be used by at least another package
+:- attr("depends_on", node(X, "gcc-runtime"), node(Y, "gcc"), "build"),
+   node_compiler(_, node(_, "gcc")),
+   not 2 { attr("depends_on", PackageNode, node(Y, "gcc"), "build") : attr("node", PackageNode) }.
+
```

Comment:
this also seems very hard-coded and needs generalization later for other runtimes.

Reply:
This is needed to avoid that the solver could pull-in a better runtime from a buildcache, if the nodes to be built are using another compiler.

Reply:
ok that's interesting.  this should probably be documented